service/history/taskProcessor.go

```diff
@@ -270,6 +270,15 @@ func (t *taskProcessor) handleTaskError(
 		return nil
 	}
 
+	if transferTask, ok := task.task.(*persistence.TransferTaskInfo); ok &&
+		transferTask.TaskType == persistence.TransferTaskTypeCloseExecution &&
+		err == ErrMissingWorkflowStartEvent &&
+		t.config.EnableDropStuckTaskByDomain(task.task.GetDomainID()) { // use domainID here to avoid accessing domainCache
+		scope.IncCounter(metrics.StuckTaskDiscardedCounter)
+		task.logger.Error("Drop close execution transfer task due to corrupted workflow history", tag.Error(err), tag.LifeCycleProcessingFailed)
```

Comment:
let's add log of workflow info for auditing

Reply:
Task logger is initialized with tags for domainID, workflowID, runID, etc.